ci/jenkins/test-rancher.sh

```diff
@@ -0,0 +1,298 @@
+    kubectl config use-context ${control_plane_node_name}
+
+    echo "=== Move kubeconfig to all Nodes ==="
+    kubectl get nodes -o wide --no-headers=true | awk '{print $6}' | while read IP; do
+        ssh -o StrictHostKeyChecking=no -n ubuntu@${IP} "if [ ! -d ".kube" ]; then mkdir .kube; fi"
+        scp -q -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -i "${WORKDIR}/.ssh/id_rsa" $KUBECONFIG_PATH ubuntu@[${IP}]:~/.kube/config
+    done
+
+    set +e
+    mkdir -p `pwd`/antrea-test-logs
+    # HACK: see https://github.com/antrea-io/antrea/issues/2292
+    go mod edit -replace github.com/moby/spdystream=github.com/antoninbas/spdystream@v0.2.1 && go mod tidy
+    go test -v antrea.io/antrea/test/e2e --logs-export-dir `pwd`/antrea-test-logs --provider remote -remote.kubeconfig="$KUBECONFIG_PATH" -timeout=100m --prometheus
+    if [[ "$?" != "0" ]]; then
+        TEST_FAILURE=true
+    fi
+    set -e
+
+    tar -zcf antrea-test-logs.tar.gz antrea-test-logs
+}
+
+function run_conformance {
+    echo "====== Running Antrea Conformance Tests ======"
+    export GO111MODULE=on
+    export GOPATH=${WORKDIR}/go
+    export GOROOT=/usr/local/go
+    export GOCACHE=${WORKDIR}/.cache/go-build
+    export PATH=$GOROOT/bin:$PATH
+
+    kubectl apply -f build/yamls/antrea.yml
+    kubectl rollout restart deployment/coredns -n kube-system
+    kubectl rollout status deployment/coredns -n kube-system
+    kubectl rollout status deployment.apps/antrea-controller -n kube-system
+    kubectl rollout status daemonset/antrea-agent -n kube-system
+
+    if [[ "$TESTCASE" =~ "conformance" ]]; then
+        ${WORKSPACE}/ci/run-k8s-e2e-tests.sh --e2e-conformance --e2e-skip "$CONFORMANCE_SKIP" --log-mode $MODE --image-pull-policy ${IMAGE_PULL_POLICY} --kube-conformance-image-version "auto" > ${WORKSPACE}/test-result.log
+    else
+        ${WORKSPACE}/ci/run-k8s-e2e-tests.sh --e2e-network-policy --e2e-skip "$NETWORKPOLICY_SKIP" --log-mode $MODE --image-pull-policy ${IMAGE_PULL_POLICY} --kube-conformance-image-version "auto" > ${WORKSPACE}/test-result.log
+    fi
+
+    cat ${WORKSPACE}/test-result.log
+    if grep -Fxq "Failed tests:" ${WORKSPACE}/test-result.log; then
+        echo "Failed cases exist."
+        TEST_FAILURE=true
+    else
+        echo "All tests passed."
+    fi
+}
+
+function clean_tmp() {
+    echo "===== Clean up stale files & folders older than 7 days under /tmp ====="
+    CLEAN_LIST=(
+        "*codecov*"
+        "kustomize-*"
+        "*antrea*"
+        "go-build*"
+    )
+    for item in "${CLEAN_LIST[@]}"; do
+        find /tmp -name "${item}" -mtime +7 -exec rm -rf {} \; 2>&1 | grep -v "Permission denied" || true
+    done
+    find ${WORKDIR} -name "support-bundles*" -mtime +7 -exec rm -rf {} \; 2>&1 | grep -v "Permission denied" || true
+}
+
+rancher_login
+
+clean_tmp
+trap clean_vm_agent EXIT
+if [[ ${TESTCASE} =~ "e2e" ]]; then
+    deliver_antrea
+    run_e2e
+else
+    deliver_antrea
```

Comment:
Looks like deliver_antrea can be extracted out, and be put before line 288 `if`.